docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/Cache.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public static class Cache
+    {
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Get a small dataset as an IEnumerable.
+            IEnumerable<SamplesUtils.DatasetUtils.FloatLabelFloatFeatureVectorSample> enumerableOfData = SamplesUtils.DatasetUtils.GenerateFloatLabelFloatFeatureVectorSamples(100000);
+            var data = mlContext.Data.ReadFromEnumerable(enumerableOfData);
+
+            // Time how long it takes to page through the records if we don't cache.
+            int lines = 0;
+            double averageOfColumn0 = 0.0;
+            var watch = System.Diagnostics.Stopwatch.StartNew();
+            var enumerable = mlContext.CreateEnumerable<SamplesUtils.DatasetUtils.FloatLabelFloatFeatureVectorSample>(data, reuseRowObject: true);
+            foreach (var row in enumerable)
+            {
+                lines++;
+                averageOfColumn0 += row.Features[0];
+            }
+            watch.Stop();
+            averageOfColumn0 /= lines;
+            var elapsedSeconds = watch.ElapsedMilliseconds / 1000.0;
+            Console.WriteLine($"Lines={lines}, averageOfColumn0={averageOfColumn0:0.00} and took {elapsedSeconds} seconds.");
```

Comment:
I suggest extracting this as a function, which is repeated 3 times. Repeating the same code will make the example look more complicated than what it actually is.

Reply:
Done.